sidsbrmnn/stripe-checkout-sessions
# Create API endpoints to create bookings

Create Next API endpoints at `/api/bookings` to initialize and verify a payment
session using Stripe.

To generate a new Payment Intent or a Subscription,

```http
POST /api/bookings/initialize
Content-Type: application/json

{
  "bathrooms": 1,
  "bedrooms": 1,
  "cleaningType": "Standard",
  "interval": "One Time"
}
```

To verify the payment and confirm the booking,

```http
POST /api/bookings/verify
Content-Type: application/json

{
  "interval": "One Time",
  "paymentId": "pi_1DpcVL2eZvKYlo2CQT4lFQr1", // use subscriptionId in case of subscriptions
  "bathrooms": 1,
  "bedrooms": 1,
  "cleaningType": "Standard"
}
```

- Install stripe and next-connect
- Create stripe helpers
- Added support for interval
- Added types for Props
- Update stripe helpers with Interval
- Create POST /bookings/initialize route
- Update calculate total price with interval
- Create POST /bookings/verify to verify payments
- Create .env.example
- Create .editorconfig
